util/progress/progressui/printer.go

```diff
@@ -28,13 +29,21 @@ type lastStatus struct {
 
 type textMux struct {
 	w         io.Writer
+	printer   vertexPrinter
```

Comment:
Ideally, the refactoring of current progress objects and new interfaces would be in a separate commit from the new JSON display. But if you are unsure yet if this is the final version of the refactors we can delay that split.